plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -70,27 +71,37 @@
     private final JdbcClient delegate;
     private final List<PropertyMetadata<?>> sessionProperties;
     private final boolean cacheMissing;
+    private final JdbcIdentityCacheMapping identityMapping;
 
-    private final Cache<JdbcIdentity, Set<String>> schemaNamesCache;
+    private final Cache<JdbcIdentityCacheKey, Set<String>> schemaNamesCache;
     private final Cache<TableNamesCacheKey, List<SchemaTableName>> tableNamesCache;
     private final Cache<TableHandleCacheKey, Optional<JdbcTableHandle>> tableHandleCache;
     private final Cache<ColumnsCacheKey, List<JdbcColumnHandle>> columnsCache;
-    private final List<PropertyMetadata<?>> sessionProperties;
     private final Cache<TableStatisticsCacheKey, TableStatistics> statisticsCache;
 
     @Inject
-    public CachingJdbcClient(@StatsCollecting JdbcClient delegate, Set<SessionPropertiesProvider> sessionPropertiesProviders, BaseJdbcConfig config)
+    public CachingJdbcClient(
```

Comment:
could you please move chopping to new lines to the first commit?

Reply:
It was not needed at that point.